code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -128,9 +131,8 @@
 			add_module(module, FALSE, FALSE)
 	for(var/obj/item/module in added_modules)
 		add_module(module, FALSE, FALSE)
-	for(var/module in held_modules)
-		if(module)
-			cyborg.equip_module_to_slot(module, held_modules.Find(module))
+	for(var/obj/item/module in held_modules & modules)
```

Comment:
```suggestion
	for(var/obj/item/module as anything in held_modules & modules)
```
By default byond will filter lists for the type of the var being iterated for, we don't want that here since it silences errors so let's disable that